### PR TITLE
Remove option to swallow reverts in Paycall and Quotecall

### PR DESCRIPTION
**Pull request by kevincheng96**
The `propagateReverts` constructor arg was introduced recently to `Paycall` and `Quotecall` to support swallowing of reverts in the two scripts. This was done to allow submitters to safely submit `Paycall/Quotecall` operations on networks without Flashbots (e.g. all L2s). Otherwise, without Flashbots' no-revert guarantee for transactions, submitters will eat the cost for any reverts in `Paycall/Quotecall`.

However, this option to swallow reverts introduces a potential gas griefing vector. When reverts are swallowed, the submitter will still get paid the payment token, at the expense of the user that signed the Quark Operation. This is normally fine and expected, since users should be paying for failing transactions and submitters need to be compensated for doing the submission. This becomes problematic when the reverting transaction is a replayable Quark Operation. In this case, a malicious submitter could repeatedly submit the reverting replayable transaction, draining the Quark wallet of its payment token in the process.

In short, we remove the option to swallow reverts in `Paycall/Quotecall` to prevent this case from happening. Submitters on networks without Flashbots are encouraged to simulate transactions first before submitting them on-chain.

### CODE REVIEW

**Comment by kevincheng96** [comment]
> But I kind of still see some scenario we might want to charge users on failed txn, like... 🤔
> 
> * User do limit order on recurring buy, but set a very tight price limit (+/- 0.0001%),  then when our system simulation passes, and submit buy txn, by the time reaching blockchain the price might no longer be valid and txn will fail on-chain.  (In this case, we might still want to charge user?)
> * User force to submit failed simulation txn (more trivial cuz we can just ban that option from user anyway)

Synced offline. Here's my response:

The reason why we are removing the ability to swallow reverts is because it opens up a very scary vector for quark wallets to be drained. If a user signs a recurring transaction that reverts, anyone can submit the transaction over and over again to drain the account and get paid via Paycall/Quotecall in the process. I think the severity of this potential attack vector is high enough to warrant removing the feature altogether, at the expense of making the submitter's job a bit more difficult for reverting txns.

**Review by hayesgm — APPROVED** [review]
